File: backend/app/dtos/recap.py
"""Оболочка документа рекапа. Внутренность разделов не типизируется намеренно:
приём не знает видов разделов, иначе он отвергал бы каждый новый вид."""
from typing import Any

from pydantic import BaseModel

from ._aliases import BODY_CONFIG

RECAP_FORMAT_VERSION = 1


class RecapDocument(BaseModel):
    model_config = BODY_CONFIG

    version: int
    book_id: int
    book: dict[str, Any]
    recap: dict[str, Any]
    retell: dict[str, Any]

File: frontend/src/api/endpoints/recap.ts
import { client } from "../client";

export type RecapSpan = { t: string; b?: boolean };

export type RecapPerson = { name: string; about: string };
export type RecapEpisode = { title: string; paragraphs: string[] };

/**
 * Section of the "Кратко" tab. `kind` decides how the section is drawn
 * (recap-document.tsx); an unfamiliar kind still shows every string found
 * inside the section, so an unknown future kind never loses content.
 */
export type RecapSection = {
  title: string;
  kind: string;
  people?: RecapPerson[];
  episodes?: RecapEpisode[];
  paragraphs?: string[];
  items?: string[];
};

export type RecapPart = { number: number; paragraphs: string[] };

export type RecapDocument = {
  version: number;
  bookId: number;
  book: { title: string; authors: string[]; series: string | null; seriesNumber: number | null };
  recap: { sections: RecapSection[] };
  retell: { parts: RecapPart[] };
};

export function fetchRecap(path: string, signal?: AbortSignal): Promise<RecapDocument> {
  return client<RecapDocument>("GET", path, { signal });
}
